### PR TITLE
Admin orders dashboard

### PR DESCRIPTION
**Pull request by jayphodges**
Additions: 
Added the ability for the admin to view and modify orders on the admin dashboard page, as well as filters for the orders on that page.

User Story:
Story 21

Concerns (code location):

Mentions to collaborators:
@djtrujillo @boveus @vaidashi 

Notes(if applicable):
Styling still needed.

### CODE REVIEW

**Comment by vaidashi** [comment]
Should add some hedge case tests soon to ensure non admin user can't alter order status.

**Comment by boveus** [comment]
I like the way we set up our post paths for changing the status of orders.  I know we could have done this in a RESTful way using update, but I think making the paths explicit rather than having the logic inside the controller makes more sense to me.

@s-espinosa @icorson3  - Do our routes for changing order statuses seem OK?  Thanks.

File: app/models/order.rb [hunk 1]
```diff
@@ -4,6 +4,11 @@ class Order < ApplicationRecord
   has_many :items, through: :item_orders
   enum status: ["ordered", "paid", "cancelled", "completed"]
 
+  scope :ordered, -> { where(status: "ordered")}
+  scope :paid, -> { where(status: "paid")}
+  scope :cancelled, -> { where(status: "cancelled")}
+  scope :completed, -> { where(status: "completed")}
```
Comment (s-espinosa) [inline app/models/order.rb:10]: Nice use of scopes here.

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -27,4 +28,19 @@ def create
...
+
+  def paid
+    Order.update(params[:order_id], status: "paid")
+    redirect_back(fallback_location: root_path)
+  end
+
+  def completed
+    Order.update(params[:order_id], status: "completed")
+    redirect_back(fallback_location: root_path)
+  end
```
Comment (s-espinosa) [inline app/controllers/orders_controller.rb:44]: I'm so on the fence about this. It's not ReSTful, but it's very clear what you're doing here, and I really like that you understand the routes file/controller#actions/controller well enough that you can implement this.

That said, in a larger application this could cause your controller to balloon as someone else in your team sees this pattern and continues to add to it as other ways to change an order are introduced. In particular, this is interesting because you're using `redirect_back`, which I don't think is what you'd want to do if you added an `order#update` controller action.

One alternative that you might consider: pass the new status as a query param. That would at least allow you to clean up some of the duplication here:

```
def update_status
  Order.update(params[:order_id], status: params[:status])
  redirect_back(fallback_location: root_path)
end
```

Alternatively, you could use the `update` controller action with the same query params, but you would need to figure out how to handle the different redirects you want to achieve.

**Review by s-espinosa — COMMENTED** [review]
@boveus Comments below. Let us know if you'd like to discuss tomorrow.
